### PR TITLE
Problem: ADR-004 does not clarify backwards incompatible changes

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added backwards incompatible changes in ADR-004 and changed its status to `Proposed`

### CODE REVIEW

File: doc/architecture/README.md [hunk 1]
```diff
@@ -29,4 +29,5 @@ To suggest an ADR, please make use of the [ADR template](./adr-template.md) prov
 | ------ | ----------- | ------ |
 | [001](./adr-001.md) | Add CosmWasm Module | Accepted |
 | [002](./adr-002.md) | Subscriptions in CosmWasm | Accepted |
-| [003](./adr-003.md) | Canis Major (1st Network Upgrade Scope of Breaking Changes) | Proposed |
\ No newline at end of file
+| [003](./adr-003.md) | Canis Major (1st Network Upgrade Scope of Breaking Changes) | Proposed |
```
Comment (tomtau) [inline doc/architecture/README.md:32]:
```suggestion
| [003](./adr-003.md) | Canis Major (1st Network Upgrade Scope of Breaking Changes) | Accepted |
```

File: doc/architecture/adr-004.md [hunk 2]
```diff
@@ -235,6 +236,12 @@ message MsgBurnNFTResponse {}
 
 To implement `MsgBurnNFT`, we can directly call the `BurnNFT` of `x/nft` (after performing validations).
 
+### Backwards incompatibility
+
+It is possible to support all the transactions and queries in a backwards compatible way except for `DenomByName` query.
+Because current implementation of NFT module specifies `denom_name` as a unique parameter unlike Cosmos SDK's native
+`x/nft` module, it is not possible to support a query which returns single `Denom` object with given `denom_name`.
```
Comment (tomtau) [inline doc/architecture/adr-004.md:243]: @kaantig is this used?

**Review by tomtau — COMMENTED** [review]
good for @kaantig to review

File: doc/architecture/README.md (hunk 1 above)
Comment (devashishdxt) [reply doc/architecture/README.md:32]: Do you want to change the status of ADR-003 as well?

File: doc/architecture/adr-004.md (hunk 2 above)
Comment (kaantig) [reply doc/architecture/adr-004.md:243]: nope, we only query it with its denom_id in the request I displayed below. we don't use denom name. 

${nodeUrl}/chainmain/nft/collections/${denomId}

File: doc/architecture/README.md (hunk 1 above)
Comment (tomtau) [reply doc/architecture/README.md:32]: yes, it's already on the mainnet / released in v2

**Review by tomtau — APPROVED** [review]

**Pull request merged**